### ISSUE TITLE
searchone mismatch between str/bytes

### ISSUE THREAD

**Issue by msimacek** [msimacek/samba#1]
````
UNEXPECTED(failure): samba.tests.core.samba.tests.core.LdbExtensionTests.test_searchone(none)
REASON: Exception: Exception: Traceback (most recent call last):
  File "/home/msimacek/git/samba/bin/python/samba/tests/core.py", line 61, in test_searchone
    l.searchone(basedn=ldb.Dn(l, "foo=dc"), attribute="bar"))
AssertionError: 'bla' != b'bla'
command: python3 -m samba.subunit.run  $LOADLIST samba.tests.core 2>&1  | /home/msimacek/git/samba/selftest/filter-subunit --fail-on-empty --prefix="samba.tests.core." --suffix="(none)"
expanded command: python3 -m samba.subunit.run  $LOADLIST samba.tests.core 2>&1  | /home/msimacek/git/samba/selftest/filter-subunit --fail-on-empty --prefix="samba.tests.core." --suffix="(none)"
ERROR: Testsuite[samba.tests.core]
REASON: Exit code was 1            
````

Mismatch between str/bytes, but I'm not sure which part of the call chain should be adjusted. `searchone` uses `ldb.schema_format_value` (implemented in C), I think that one should return str (it takes str arguments)

**Comment by encukou** [comment]
`ldb.schema_format_value` might return binary data in some cases, so it can't decode into a Unicode string. (As for arguments: the attribute name is text but value is bytes.)
However, `searchone` is apparently only used for text data (distinguishedName is Unicode/UTF-8; not sure about `objectGUID` but I'd guess it's text too). So I think the decoding should be done there.

**Comment by encukou** [comment]
Fixed in 19e682b84018d2c4bf23ebc5250eed679f5fd12c `searchone: Decode to str on Python 3`
